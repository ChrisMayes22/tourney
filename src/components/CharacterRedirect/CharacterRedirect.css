@import url('https://fonts.googleapis.com/css?family=Josefin+Sans:300|Pacifico');

.background{
    background-image: url('../../assets/background__rose-pattern.jpg');;
    height: 100vh;
    width: 100vw;
    display: flex;
    align-items: center;
    align-content: center;
    justify-content: center;
}

.ui{
    width:  80%;
    height: 80%;
    background-color: #a63939;
    color: rgb(253, 211, 250);
    font-size: 2rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    align-content: center;
    justify-content: center;
    box-shadow: 5px 4px 3px rgba(0, 0, 0, 0.7);
    border-radius: 10px;
}

[type="number"]{
    background-color: rgb(255, 239, 254);
    color: #a63939;
    text-align: center;
    height: 15%;
    width: 15%;
    font-size: 2.5rem;
    font-family: 'Pacifico', cursive;
    border: none;
}

