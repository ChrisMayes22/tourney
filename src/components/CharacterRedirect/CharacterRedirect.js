import React from 'react';
import classes from './CharacterRedirect.css';
import SubmitButton from '../SubmitButton/SubmitButton';
import { Link } from 'react-router-dom';

const characterRedirect = () => {
    return(
        <div className={classes.background}>
            <nav className={classes.ui}>
                <h3>Each players should upload exactly one character per round.</h3>
                <Link to="/upload-page">
                    <SubmitButton children={"OK"}/>
                </Link>
            </nav>
        </div>
    )
}

export default characterRedirect;