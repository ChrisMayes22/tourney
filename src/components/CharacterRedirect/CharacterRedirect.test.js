import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import CharacterRedirect from './CharacterRedirect';

describe('Basic render tests for component', function(){
    test('App renders as expected', () => {
      const wrapper = shallow(<CharacterRedirect/>)
    
      expect(wrapper).toMatchSnapshot();
    })
  })