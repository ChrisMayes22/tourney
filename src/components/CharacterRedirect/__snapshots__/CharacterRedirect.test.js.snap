// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render tests for component App renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <characterRedirect />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <h3>
          Each players should upload exactly one character per round.
        </h3>
        <Link
          replace={false}
          to="/upload"
        >
          <submitButton>
            OK
          </submitButton>
        </Link>
      </nav>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Each players should upload exactly one character per round.
          </h3>,
          <Link
            replace={false}
            to="/upload"
          >
            <submitButton>
              OK
            </submitButton>
          </Link>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Each players should upload exactly one character per round.",
          },
          "ref": null,
          "rendered": "Each players should upload exactly one character per round.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <submitButton>
              OK
            </submitButton>,
            "replace": false,
            "to": "/upload",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "OK",
            },
            "ref": null,
            "rendered": "OK",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <h3>
            Each players should upload exactly one character per round.
          </h3>
          <Link
            replace={false}
            to="/upload"
          >
            <submitButton>
              OK
            </submitButton>
          </Link>
        </nav>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Each players should upload exactly one character per round.
            </h3>,
            <Link
              replace={false}
              to="/upload"
            >
              <submitButton>
                OK
              </submitButton>
            </Link>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Each players should upload exactly one character per round.",
            },
            "ref": null,
            "rendered": "Each players should upload exactly one character per round.",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <submitButton>
                OK
              </submitButton>,
              "replace": false,
              "to": "/upload",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "OK",
              },
              "ref": null,
              "rendered": "OK",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
